web/src/message_list_data.js

```diff
@@ -259,6 +263,7 @@ export class MessageListData {
     }
 
     update_items_for_muting() {
+        message_util.render_old_messages_for_stream_and_topic_unmuting();
```

Comment:
This is almost certainly not something we can merge; the data module should never call into a rendering module, and also this seems expensive.

You'll need to read the call chain and think of a better place to manage this -- code in this module should generally only be responsible for updating data structures list `_items` and not call anything that touches the DOM at all.

Reply:
I looked deeply into the codebase and realized that the main problem lies in the fact that `this._items` does not have messages to render for a muted stream in case of unmuting. Rather than implementing an expensive fix, a simple fix is to just re-render the home-view upon unmuting the topic. I have updated the PR with this change. Could you please review it? Thanks.

Reply:
Sure, thanks for the information. I have changed the placement of this call, since we can't avoid using this function to live-update the messages of `unmuted topics` in the `All messages` feed. I have deeply explained the reasoning for this in commit messages.